apps/worker/tasks/base.py

```diff
@@ -49,6 +50,14 @@
 
 log = logging.getLogger("worker")
 
+
+# Sentinel class to distinguish "not provided" from "explicitly None" for max_retries
+class _NotProvided:
+    """Sentinel class to distinguish 'not provided' from 'explicitly None' for max_retries"""
+
+
+_NOT_PROVIDED = _NotProvided()
```

Comment:
@thomasrockhu-codecov I don't understand why a named class is not the same as a unnamed class. This is different from how I do singletons in JavaScript. Also, while the comment says "singleton", is that really what we are doing here? The idea is to check for "set vs not-set", because Celery treats None differently from "not set".